src/Compilers/CSharp/Portable/Symbols/MethodSymbol.cs

```diff
@@ -1199,6 +1199,12 @@ ImmutableArray<AttributeData> IMethodSymbol.GetReturnTypeAttributes()
         /// </summary>
         internal virtual void AddSynthesizedReturnTypeAttributes(ref ArrayBuilder<SynthesizedAttributeData> attributes)
         {
+            if (this.ReturnsByRefReadonly)
+            {
+                // PROTOTYPE(readonlyRefs) it is optional now as it will be generated in the next PR
+                var constructor = WellKnownMember.System_Runtime_CompilerServices_ReadOnlyAttribute__ctor;
+                AddSynthesizedAttribute(ref attributes, this.DeclaringCompilation.TrySynthesizeAttribute(constructor, isOptionalUse: true));
```

Comment:
@cston alright. I'll extract to a helper method and remove the local. Thanks!

---
In reply to: [110692798](https://github.com/dotnet/roslyn/pull/18358#discussion_r110692798) [](ancestors = 110692798)